src/main/oauth2/utils.ts

```diff
@@ -0,0 +1,148 @@
+import { OAUTH_AUTHORIZATION_SERVER_URL, OAUTH_CLIENT_ID, WELL_KNOWN_OAUTH_AUTHORIZATION_SERVER_URL } from "@main/config/server";
+import { generatePKCE } from "@main/oauth2/pkce";
+import RedirectHandler from "@main/oauth2/redirect-handler";
+import { logger } from "@main/utils/logger";
+import { shell } from "electron";
+import { stringify } from "querystring";
+
+const log = logger("oauth2-utils");
+
+interface TokenResponse {
+    token: string;
+    refreshToken: string;
+    expiresIn: number;
+}
+
+export async function fetchAuthorizationServerMetadata(): Promise<{
+    authorizationEndpoint: string;
+    tokenEndpoint: string;
+}> {
+    const response = await fetch(WELL_KNOWN_OAUTH_AUTHORIZATION_SERVER_URL);
+    if (response.status !== 200) {
+        const error = `Failed to fetch OAuth2 authorization server metadata: ${response.status} ${response.statusText}`;
+        log.error(error);
+        throw new Error(error);
+    }
+    const { authorization_endpoint, token_endpoint } = await response.json();
+    if (!authorization_endpoint || !token_endpoint) {
+        const error = "Invalid OAuth2 authorization server metadata";
+        log.error(error);
+        throw new Error(error);
+    }
+    // TODO: Remove this hack once the server is fixed
```

Comment:
Add the link: https://github.com/beyond-all-reason/teiserver/pull/555 it is, at time of writing, merged, but not yet deployed.